src/test/java/com/google/api/codegen/testdata/csharp_gapic_snippets_library.baseline

```diff
@@ -0,0 +1,696 @@
+            // End snippet
+        }
+
+        public void MergeShelves()
+        {
+            // Snippet: MergeShelves(string,string,CallSettings)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatShelfName("[SHELF]");
+            string formattedOtherShelfName = LibraryServiceClient.FormatShelfName("[SHELF]");
+            // Make the request
+            Shelf response = libraryServiceClient.MergeShelves(formattedName, formattedOtherShelfName);
+            // End snippet
+        }
+
+        public async Task CreateBookAsync()
+        {
+            // Snippet: CreateBookAsync(string,Book,CallSettings)
+            // Additional: CreateBookAsync(string,Book,CancellationToken)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatShelfName("[SHELF]");
+            Book book = new Book();
+            // Make the request
+            Book response = await libraryServiceClient.CreateBookAsync(formattedName, book);
+            // End snippet
+        }
+
+        public void CreateBook()
+        {
+            // Snippet: CreateBook(string,Book,CallSettings)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            string formattedName = LibraryServiceClient.FormatShelfName("[SHELF]");
+            Book book = new Book();
+            // Make the request
+            Book response = libraryServiceClient.CreateBook(formattedName, book);
+            // End snippet
+        }
+
+        public async Task PublishSeriesAsync()
+        {
+            // Snippet: PublishSeriesAsync(Shelf,IEnumerable<Book>,uint,CallSettings)
+            // Additional: PublishSeriesAsync(Shelf,IEnumerable<Book>,uint,CancellationToken)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            Shelf shelf = new Shelf();
+            IEnumerable<Book> books = new List<???>();
+            uint edition = 0;
+            // Make the request
+            PublishSeriesResponse response = await libraryServiceClient.PublishSeriesAsync(shelf, books, edition);
+            // End snippet
+        }
+
+        public void PublishSeries()
+        {
+            // Snippet: PublishSeries(Shelf,IEnumerable<Book>,uint,CallSettings)
+            // Create client
+            LibraryServiceClient libraryServiceClient = LibraryServiceClient.Create();
+            // Initialize request argument(s)
+            Shelf shelf = new Shelf();
+            IEnumerable<Book> books = new List<???>();
```

Comment:
Is this one of the missing generic bits? I haven't looked at the code yet, but we seem to know the types...